TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -23,6 +24,7 @@ public CrossCompleteChecker([NotNull]TypeCobolOptions compilerOptions)
         }
 
         private readonly TypeCobolOptions _compilerOptions;
+        private Node _currentNodeScope;
```

Comment:
Ok I'm not a huge fan of the name because to actually understand what this field refers to, you have to read the code where it is used. How about replacing that with a reference to the current section ? It would be null when the paragraph is declared directly in procedure division and set to the current section in `Visit(Section)` method like you did. Storing the `procedure division` node does not give anything and `private Node _currentSection` is clearer.

Reply:
Done